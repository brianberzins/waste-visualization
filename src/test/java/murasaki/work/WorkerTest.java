package murasaki.work;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class WorkerTest {

    @Test
    void testWorker() {
        var worker = new SimpleWorker();
        assertTrue(worker.addWork(new Work(Duration.ofHours(8))));
        assertEquals(Duration.ofHours(3), worker.perform(Duration.ofHours(3)));
        assertTrue(worker.addWork(new Work(Duration.ofHours(9))));
        assertEquals(Duration.ofHours(7), worker.perform(Duration.ofHours(7)));
        assertEquals(Duration.ofHours(7), worker.perform(Duration.ofHours(12)));
        Approvals.verifyAll("work", worker.workList);
    }
}
