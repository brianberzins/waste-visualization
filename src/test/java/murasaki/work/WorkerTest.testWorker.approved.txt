work[0] = remaining=PT0S
history=[
  WORK PT8H
]
work[1] = remaining=PT0S
history=[
  WAIT PT5H,
  WORK PT9H
]
